src/range_parameters.rs

```diff
@@ -109,7 +112,7 @@ where P: FromUniformBytes + Compressable + Clone
 
     /// Return the non-public value bulletproof generator references
     pub fn hi_base_copied(&self) -> Vec<P> {
-        self.hi_base_iter().cloned().collect()
+        self.hi_base_iter().copied().collect()
```

Comment:
I made an issue about the memory cloning in BP+, this is one of them, is there a reason we copy it here and not just return a ref to it?
If the callee really wants to clone it, they can. Here it is forced upon the callee.

Reply:
Removed in a new update.